upgrade/crd.yaml

```diff
@@ -1,68 +1,362 @@
+          type: string
+        metadata:
+          type: object
+        spec:
+          description: VolumeInfo contains the volume related info for all types of
+            volumes in ZFSVolume
+          properties:
+            capacity:
+              description: Capacity of the volume
+              minLength: 1
+              type: string
+            compression:
+              description: Controls the compression algorithm used for this dataset.
+                Compression specifies if the it should enabled on the zvol. Setting
+                compression to on indicates that the current default compression algorithm
+                should be used. The current default compression algorithm is either
+                lzjb or, if the lz4_compress feature is enabled, lz4. Changing this
+                property affects only newly-written data.
+              pattern: ^(on|off|lzjb|gzip|gzip-[1-9]|zle|lz4)$
+              type: string
+            dedup:
+              description: Deduplication is the process for removing redundant data
+                at the block level, reducing the total amount of data stored. If a
+                file system has the dedup property enabled, duplicate data blocks
+                are removed synchronously. The result is that only unique data is
+                stored and common components are shared among files. Deduplication
+                can consume significant processing power (CPU) and memory as well
+                as generate additional disk IO. Before creating a pool with deduplication
+                enabled, ensure that you have planned your hardware requirements appropriately
+                and implemented appropriate recovery practices, such as regular backups.
+                As an alternative to deduplication consider using compression=lz4,
+                as a less resource-intensive alternative. should be enabled on the
+                zvol
+              enum:
+              - "on"
+              - "off"
+              type: string
+            encryption:
+              description: Enabling the encryption feature allows for the creation
+                of encrypted filesystems and volumes. ZFS will encrypt file and zvol
+                data, file attributes, ACLs, permission bits, directory listings,
+                FUID mappings, and userused / groupused data. ZFS will not encrypt
+                metadata related to the pool structure, including dataset and snapshot
+                names, dataset hierarchy, properties, file size, file holes, and deduplication
+                tables (though the deduplicated data itself is encrypted).
+              pattern: ^(on|off|aes-128-[c,g]cm|aes-192-[c,g]cm|aes-256-[c,g]cm)$
+              type: string
+            fsType:
+              description: FsType specifies filesystem type for the zfs volume/dataset
```

Comment:
any default value, probably can get into details of what happens if filesystem is specified as zfs vs xfs/ext4.